src/vehicle_use.cpp

```diff
@@ -1485,7 +1485,14 @@ veh_interact_results vehicle::interact_with( const tripoint &pos, int interact_p
         if( fuel_left( "battery" ) < pseudo.ammo_required() ) {
             return false;
         }
-        auto qty = pseudo.ammo_capacity() - discharge_battery( pseudo.ammo_capacity() );
+        auto capacity = 0;
+        if( pseudo.magazine_default() != "null" ) {
+            item mag( pseudo.magazine_default() );
```

Comment:
Instantiating the whole `item` for accessing its `type` is a bit wasteful. You can use `item::find_type` for this.

Reply:
Thank you very for telling me this. Because C++ is so alien to me, I'm usually just blindly fumbling for a bit of code that appears to do approximately what i want, and I sometimes overlook things that might work because I have difficulty discerning what is interrelated how. I'll be making use of that going forward.

Reply:
Made use of `item::find_type` in the logic in `item::ammo_capacity()` (since the logic is no longer in `vehicle_use.cpp`)